lib/aggregate/container.rb

```diff
@@ -5,21 +5,30 @@
 module Aggregate
   module Container
     extend ActiveSupport::Concern
-    include LargeTextField::Owner
     include Aggregate::AggregateStore
+    include ActiveSupport::Callbacks
 
     included do
-      large_text_field :aggregate_store
-      set_callback(:large_text_field_save, :before, :write_aggregates)
       validate :validate_aggregates
       send(:define_callbacks, :aggregate_load)
       send(:define_callbacks, :aggregate_load_schema)
       send(:define_callbacks, :aggregate_load_check_schema)
-      class_attribute :aggregate_container_options
-      self.aggregate_container_options = {
-        use_storage_field: nil,
-        use_large_text_field_as_failover: false
-      }
+      class_attribute :aggregate_storage_field
+      class_attribute :migrate_from_storage_field
+
+      def self.store_aggregates_using_large_text_field
+        include LargeTextField::Owner
+        large_text_field :aggregate_store
+        self.aggregate_storage_field = :aggregate_store
+        self.migrate_from_storage_field = nil
+        set_callback(:large_text_field_save, :before, :write_aggregates)
+      end
+
+      def self.store_aggregates_using(storage_field, migrate_from_storage_field: nil)
+        self.aggregate_storage_field = storage_field
+        self.migrate_from_storage_field = migrate_from_storage_field
+        set_callback(:save, :before, :write_aggregates)
+      end
```

Comment:
Should `store_aggregates_using_large_text_field` and `store_aggregates_using` have checks to ensure they're not both being used in the same class?
e.g.
```
self.aggregate_storage_field and raise "aggregate_storage_field is already set to #{self.aggregate_storage_field.inspect}"
self.aggregate_storage_field = storage_field